### PR TITLE
KOI should first try to terminate a service with SIGTERM

### PR DESCRIPTION
**Pull request by abhayyb**
before a SIGKILL. This will help the child process to
handle the SIGTERM and clean up stale files and terminate
their child processes.

### CODE REVIEW

**Comment by krig** [comment]
Sure, looks good to me.

**Comment by krig** [comment]
Hmm. I guess the only thing would be the hardcoded 1000*1000 timeouts. At least making it a constant so that it doesn't have to be updated in multiple places would make sense.

**Comment by abhayyb** [comment]
Made the changes as per review comments.

File: src/servicemgr.cpp [hunk 1]
```diff
@@ -526,13 +527,11 @@ namespace koi {
...
+						s._running.termkill(TERMINATE_TIMEOUT);
 						s.unrun();
-						if (s._state == Svc_Starting) {
-							s.transition(Svc_Started);
-							LOG_WARN("Forcing stop");
-							s.stop(_logproxy.inpipe);
-						}
+						s.transition(Svc_Started);
+						LOG_WARN("Forcing stop");
+						s.stop(_logproxy.inpipe);
```
Comment (krig) [inline src/servicemgr.cpp]: What is the reason for this change? I don't remember this part from the earlier diff..

Comment (abhayyb) [reply src/servicemgr.cpp]: This change was there earlier too.
The condition check `if (s._state == Svc_Starting)` is duplicated.
Also `s._state` didn't change in between.
So removed this redundant check.

Comment (krig) [reply src/servicemgr.cpp]: Ah yes, I see now. Thanks!
